src/kixi/hecuba/onyx/jobs/measurements_connector.clj

```diff
@@ -2,14 +2,49 @@
   (:require [cheshire.core :as json]
             [clj-http.client :as client]
             [environ.core :refer [env]]
-            [schema.core :as s]))
+            [schema.core :as s]
+            [taoensso.timbre :as timbre]))
+
+
+(defn push-payload-to-hecuba
+  "Create the http post request for measurements
+  uploads"
+  [json-payload entity-id device-id]
+  (let [json-to-send (json/generate-string {:measurements json-payload})
+        endpoint (str (env :hecuba-endpoint) "entities/" entity-id "/devices/" device-id "/measurements/")]
+
+    (timbre/info (format "DATA: %s" json-to-send))
+
+
+    (timbre/info "Using endpoint: %s" (str (env :hecuba-endpoint) "entities/" entity-id "/devices/" device-id "/measurements/"))
+
+    (try (client/post
+          endpoint
+          {:basic-auth [(env :hecuba-username) (env :hecuba-password)]
+           :body json-to-send
+           :headers {"X-Api-Version" "2"}
+           :content-type :json
+           :socket-timeout 20000
+           :conn-timeout 20000
+           :accept "application/json"})
+         (catch Exception e (doall (str "Caught Exception " (.getMessage e))
+                                   (timbre/error e "> There was an error during the upload to entity " entity-id)))
+         (finally {:message "push-payload-to-hecuba complete."}))))
 
 (defn get-data [fn-data]
   ;; map of data passed in from the workflow here.
   ;; TODO - needs to take the measurements and save them via Hecuba API
-
-  (println (str "k.h.o.j.hf - data - " fn-data)))
-
+  (let [entity-id (get-in fn-data [:kafka-payload :entity-id])
+        device-id (get-in fn-data [:kafka-payload :device_id])
+        measurements (:measurements fn-data)
+        degree-day [(:degree-day fn-data)]]
+    (if (> (count measurements) 0)
+      (push-payload-to-hecuba measurements entity-id device-id)
+      (timbre/info (format "Writing measurements to device %s on entity %s" device-id entity-id)))
+    (if (> (count degree-day) 0)
+      (push-payload-to-hecuba degree-day entity-id device-id)
+      (timbre/info (format "Writing degree day reading to device %s on entity %s" device-id entity-id))))
+  {:done true})
 
```

Comment:
Just to confirm Tom, the println/timbre changes were already done. So I'm happy to merge.